### PR TITLE
Replace old `%-formatted` by `f-strings`

### PR DESCRIPTION
**Pull request by hussein-awala**
Replace old %-formatted by f-strings and add `flynt` hook to pre-commit to auto-detect and fix them.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -70,3 +70,11 @@ repos:
           ]
         additional_dependencies:
           - tomli==2.0.1
+  - repo: https://github.com/ikamensh/flynt
```
Comment (Fokko) [inline .pre-commit-config.yaml:73]: Instead of adding a new plugin, can you enable the check in Ruff? https://docs.astral.sh/ruff/rules/#flynt-fly

**Review by Fokko — COMMENTED** [review]
Great suggestion @hussein-awala 🙌

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hussein-awala) [reply .pre-commit-config.yaml:73]: Yes, I will check it.

Comment (hussein-awala) [reply .pre-commit-config.yaml:73]: The implementation of Flynt in ruff is still in progress (https://github.com/astral-sh/ruff/issues/2102), the current rule doesn't detect all the usages of the old styles, and I cannot provide the argument `line-length` to deal with long lines. WDYT?

Comment (Fokko) [reply .pre-commit-config.yaml:73]: Thanks, I was unaware of that. I ran it locally, and it didn't pick up the violations that are caught here.

**Review by Fokko — APPROVED** [review]

**Pull request merged**